### PR TITLE
update tidb_opt_fix_control for v6.5

### PR DESCRIPTION
**Pull request by qw4990**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -2475,6 +2475,17 @@ mysql> desc select count(distinct a) from test.t;
 - 控制优化器是否会选择表的哈希连接。默认打开 (`ON`)。设置为 `OFF` 时，优化器在生成执行计划时会避免选择表的哈希连接，除非没有其他连接方式可用。
 - 如果同时使用了 `tidb_opt_enable_hash_join` 和 `HASH_JOIN` Hint，则 `HASH_JOIN` Hint 优先级更高。即使 `tidb_opt_enable_hash_join` 被设置为 `OFF`，如果在查询中指定了 `HASH_JOIN` Hint，TiDB 优化器仍然会强制执行哈希连接计划。
 
+### `tidb_opt_fix_control` <span class="version-mark">从 v6.5.7 和 v7.1.0 版本开始引入</span>
```
Comment (Oreoxmt) [inline system-variables.md:2478]:
```suggestion
### `tidb_opt_fix_control` <span class="version-mark">从 v6.5.7 版本开始引入</span>
```

File: optimizer-fix-controls.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+随着产品迭代演进，TiDB 优化器的行为会发生变化，进而生成更加合理的执行计划。但在某些特定场景下，新的行为可能会导致非预期结果。例如：
+
+- 部分行为的效果和场景相关。有的行为改变，能在大多数场景下带来改进，但可能在极少数场景下导致回退。
+- 有时，行为细节的变化和其导致的结果之间的关系十分复杂。即使是对某处行为细节的改进，也可能在整体上导致执行计划回退。
+
+因此，TiDB 提供了 Optimizer Fix Controls 功能，允许用户通过设置一系列 Fix 控制 TiDB 优化器的行为细节。本文档介绍了 Optimizer Fix Controls 及其使用方法，并列举了当前 TiDB 支持调整的所有 Fix。
+
+## `tidb_opt_fix_control` 介绍
+
+从 TiDB v7.1.0 开始，提供了 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v657-和-v710-版本开始引入) 系统变量来更细粒度地控制优化器的行为。
```
Comment (Oreoxmt) [inline optimizer-fix-controls.md:17]:
```suggestion
从 TiDB v7.1.0 开始，提供了 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v657-版本开始引入) 系统变量来更细粒度地控制优化器的行为。
```

File: optimizer-fix-controls.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+`tidb_opt_fix_control` 支持设置多个 Fix，不同 Fix 之间使用逗号 (`,`) 分隔。格式形如 `"<#issue1>:<value1>,<#issue2>:<value2>,...,<#issueN>:<valueN>"`，其中 `<#issueN>` 代表 Fix 编号。例如：
+
+```sql
+SET SESSION tidb_opt_fix_control = '44262:ON,44389:ON';
+```
+
+## Optimizer Fix Controls 参考
+
+### [`44262`](https://github.com/pingcap/tidb/issues/44262) <span class="version-mark">从 v7.2.0 版本开始引入</span>
```
Comment (Oreoxmt) [inline optimizer-fix-controls.md:29]:
```suggestion
### [`44262`](https://github.com/pingcap/tidb/issues/44262) <span class="version-mark">从 v6.5.7 版本开始引入</span>
```

File: optimizer-fix-controls.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+- 默认值：`OFF`
+- 可选值：`ON`、`OFF`
+- 是否允许在缺少 [GlobalStats](/statistics.md#动态裁剪模式下的分区表统计信息) 的情况下使用[动态裁剪模式](/partitioned-table.md#动态裁剪模式)访问分区表。
+
+### [`44389`](https://github.com/pingcap/tidb/issues/44389) <span class="version-mark">从 v7.2.0 版本开始引入</span>
```
Comment (Oreoxmt) [inline optimizer-fix-controls.md:35]:
```suggestion
### [`44389`](https://github.com/pingcap/tidb/issues/44389) <span class="version-mark">从 v6.5.7 版本开始引入</span>
```

File: optimizer-fix-controls.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+- 默认值：`OFF`
+- 可选值：`ON`、`OFF`
+- 对形如 `c = 10 and (a = 'xx' or (a = 'kk' and b = 1))` 的过滤条件，是否尝试为 `IndexRangeScan` 更加完整地构造扫描范围，即 `range`。
+
+### [`44830`](https://github.com/pingcap/tidb/issues/44830) <span class="version-mark">从 v7.3.0 版本开始引入</span>
```
Comment (Oreoxmt) [inline optimizer-fix-controls.md:41]:
```suggestion
### [`44830`](https://github.com/pingcap/tidb/issues/44830) <span class="version-mark">从 v6.5.7 版本开始引入</span>
```

File: optimizer-fix-controls.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+- 默认值：`OFF`
+- 可选值：`ON`、`OFF`
+- 此开关控制是否让 Plan Cache 对在物理优化阶段形成的 `PointGet` 计划进行缓存。
+
+### [`44855`](https://github.com/pingcap/tidb/issues/44855) <span class="version-mark">从 v7.3.0 版本开始引入</span>
```
Comment (Oreoxmt) [inline optimizer-fix-controls.md:47]:
```suggestion
### [`44855`](https://github.com/pingcap/tidb/issues/44855) <span class="version-mark">从 v6.5.7 版本开始引入</span>
```

File: optimizer-fix-controls.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+随着产品迭代演进，TiDB 优化器的行为会发生变化，进而生成更加合理的执行计划。但在某些特定场景下，新的行为可能会导致非预期结果。例如：
+
+- 部分行为的效果和场景相关。有的行为改变，能在大多数场景下带来改进，但可能在极少数场景下导致回退。
+- 有时，行为细节的变化和其导致的结果之间的关系十分复杂。即使是对某处行为细节的改进，也可能在整体上导致执行计划回退。
+
+因此，TiDB 提供了 Optimizer Fix Controls 功能，允许用户通过设置一系列 Fix 控制 TiDB 优化器的行为细节。本文档介绍了 Optimizer Fix Controls 及其使用方法，并列举了当前 TiDB 支持调整的所有 Fix。
+
+## `tidb_opt_fix_control` 介绍
+
+从 TiDB v7.1.0 开始，提供了 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v657-版本开始引入) 系统变量来更细粒度地控制优化器的行为。
```
Comment (Oreoxmt) [inline optimizer-fix-controls.md:17]:
```suggestion
从 TiDB v6.5.7 开始，提供了 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v657-版本开始引入) 系统变量来更细粒度地控制优化器的行为。
```

**Review by time-and-fate — COMMENTED** [review]
`tidb_opt_fix_control` 是在 v6.5.3 和 fix 44262 一起被 cherry-pick 回来的。另外的几个 fix 是在不同的小版本 cherry-pick 回来的。要不要区分一下？

File: control-execution-plan.md [hunk 8]
```diff
@@ -8,4 +8,4 @@ SQL 性能调优的前两个章节介绍了如何理解 TiDB 的执行计划以
 
 - [Optimizer Hints](/optimizer-hints.md)中，我们会介绍如何使用 Hint 来指导 TiDB 生成执行计划。
 - 但是使用 Hint 会侵入性地更改 SQL，在一些场景下并不能简单的插入 Hint。在[执行计划管理](/sql-plan-management.md)中，我们会介绍 TiDB 如何使用另一种语法来非侵入地控制执行计划的生成，同时还会介绍后台自动对执行计划进行演进的手段。该手段可用来减轻诸如版本升级等原因造成的执行计划不稳定和集群性能下降的问题。
-- 最后在[优化规则及表达式下推的黑名单](/blocklist-control-plan.md)中，我们会介绍黑名单的使用。
+- 最后在[优化规则及表达式下推的黑名单](/blocklist-control-plan.md)中，我们会介绍黑名单的使用。详细介绍请参考 [Optimizer Fix Controls](/optimizer-fix-controls.md)。
```
Comment (Oreoxmt) [inline control-execution-plan.md:11]:
```suggestion
- 最后在[优化规则及表达式下推的黑名单](/blocklist-control-plan.md)中，我们会介绍黑名单的使用。

除以上手段之外，执行计划还会被一部分系统变量所影响。通过在系统级或会话级对变量进行修改，能够控制执行计划的生成。自 v6.5.7 起，TiDB 引入了一个相对特殊的变量 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v657-版本开始引入)。这个变量能够接受多个控制项，用来更细粒度地控制优化器的行为，避免集群升级后优化器行为变化导致的性能回退。详细介绍请参考 [Optimizer Fix Controls](/optimizer-fix-controls.md)。
```

**Review by Oreoxmt — APPROVED** [review]

File: system-variables.md [hunk 9]
```diff
@@ -2475,6 +2475,17 @@ mysql> desc select count(distinct a) from test.t;
 - 控制优化器是否会选择表的哈希连接。默认打开 (`ON`)。设置为 `OFF` 时，优化器在生成执行计划时会避免选择表的哈希连接，除非没有其他连接方式可用。
 - 如果同时使用了 `tidb_opt_enable_hash_join` 和 `HASH_JOIN` Hint，则 `HASH_JOIN` Hint 优先级更高。即使 `tidb_opt_enable_hash_join` 被设置为 `OFF`，如果在查询中指定了 `HASH_JOIN` Hint，TiDB 优化器仍然会强制执行哈希连接计划。
 
+### `tidb_opt_fix_control` <span class="version-mark">从 v6.5.7 版本开始引入</span>
```
Comment (Oreoxmt) [inline system-variables.md:2478]:
```suggestion
### `tidb_opt_fix_control` <span class="version-mark">从 v6.5.3 版本开始引入</span>
```

File: optimizer-fix-controls.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+随着产品迭代演进，TiDB 优化器的行为会发生变化，进而生成更加合理的执行计划。但在某些特定场景下，新的行为可能会导致非预期结果。例如：
+
+- 部分行为的效果和场景相关。有的行为改变，能在大多数场景下带来改进，但可能在极少数场景下导致回退。
+- 有时，行为细节的变化和其导致的结果之间的关系十分复杂。即使是对某处行为细节的改进，也可能在整体上导致执行计划回退。
+
+因此，TiDB 提供了 Optimizer Fix Controls 功能，允许用户通过设置一系列 Fix 控制 TiDB 优化器的行为细节。本文档介绍了 Optimizer Fix Controls 及其使用方法，并列举了当前 TiDB 支持调整的所有 Fix。
+
+## `tidb_opt_fix_control` 介绍
+
+从 TiDB v6.5.7 开始，提供了 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v657-版本开始引入) 系统变量来更细粒度地控制优化器的行为。
```
Comment (Oreoxmt) [inline optimizer-fix-controls.md:17]:
```suggestion
从 TiDB v6.5.3 开始，提供了 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v653-版本开始引入) 系统变量来更细粒度地控制优化器的行为。
```

File: control-execution-plan.md [hunk 11]
```diff
@@ -9,3 +9,5 @@ SQL 性能调优的前两个章节介绍了如何理解 TiDB 的执行计划以
 - [Optimizer Hints](/optimizer-hints.md)中，我们会介绍如何使用 Hint 来指导 TiDB 生成执行计划。
 - 但是使用 Hint 会侵入性地更改 SQL，在一些场景下并不能简单的插入 Hint。在[执行计划管理](/sql-plan-management.md)中，我们会介绍 TiDB 如何使用另一种语法来非侵入地控制执行计划的生成，同时还会介绍后台自动对执行计划进行演进的手段。该手段可用来减轻诸如版本升级等原因造成的执行计划不稳定和集群性能下降的问题。
 - 最后在[优化规则及表达式下推的黑名单](/blocklist-control-plan.md)中，我们会介绍黑名单的使用。
+
+除以上手段之外，执行计划还会被一部分系统变量所影响。通过在系统级或会话级对变量进行修改，能够控制执行计划的生成。自 v6.5.7 起，TiDB 引入了一个相对特殊的变量 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v657-版本开始引入)。这个变量能够接受多个控制项，用来更细粒度地控制优化器的行为，避免集群升级后优化器行为变化导致的性能回退。详细介绍请参考 [Optimizer Fix Controls](/optimizer-fix-controls.md)。
```
Comment (Oreoxmt) [inline control-execution-plan.md:13]:
```suggestion
除以上手段之外，执行计划还会被一部分系统变量所影响。通过在系统级或会话级对变量进行修改，能够控制执行计划的生成。自 v6.5.3 起，TiDB 引入了一个相对特殊的变量 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v657-版本开始引入)。这个变量能够接受多个控制项，用来更细粒度地控制优化器的行为，避免集群升级后优化器行为变化导致的性能回退。详细介绍请参考 [Optimizer Fix Controls](/optimizer-fix-controls.md)。
```

File: optimizer-fix-controls.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+`tidb_opt_fix_control` 支持设置多个 Fix，不同 Fix 之间使用逗号 (`,`) 分隔。格式形如 `"<#issue1>:<value1>,<#issue2>:<value2>,...,<#issueN>:<valueN>"`，其中 `<#issueN>` 代表 Fix 编号。例如：
+
+```sql
+SET SESSION tidb_opt_fix_control = '44262:ON,44389:ON';
+```
+
+## Optimizer Fix Controls 参考
+
+### [`44262`](https://github.com/pingcap/tidb/issues/44262) <span class="version-mark">从 v6.5.7 版本开始引入</span>
```
Comment (Oreoxmt) [inline optimizer-fix-controls.md:29]:
```suggestion
### [`44262`](https://github.com/pingcap/tidb/issues/44262) <span class="version-mark">从 v6.5.3 版本开始引入</span>
```

File: optimizer-fix-controls.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+
+- 默认值：`OFF`
+- 可选值：`ON`、`OFF`
+- 是否允许在缺少 [GlobalStats](/statistics.md#动态裁剪模式下的分区表统计信息) 的情况下使用[动态裁剪模式](/partitioned-table.md#动态裁剪模式)访问分区表。
+
+### [`44389`](https://github.com/pingcap/tidb/issues/44389) <span class="version-mark">从 v6.5.7 版本开始引入</span>
```
Comment (Oreoxmt) [inline optimizer-fix-controls.md:35]:
```suggestion
### [`44389`](https://github.com/pingcap/tidb/issues/44389) <span class="version-mark">从 v6.5.3 版本开始引入</span>
```

File: optimizer-fix-controls.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+- 可选值：`ON`、`OFF`
+- 对形如 `c = 10 and (a = 'xx' or (a = 'kk' and b = 1))` 的过滤条件，是否尝试为 `IndexRangeScan` 更加完整地构造扫描范围，即 `range`。
+
+### [`44830`](https://github.com/pingcap/tidb/issues/44830) <span class="version-mark">从 v6.5.7 版本开始引入</span>
+
+- 默认值：`OFF`
+- 可选值：`ON`、`OFF`
+- 此开关控制是否让 Plan Cache 对在物理优化阶段形成的 `PointGet` 计划进行缓存。
+
+### [`44855`](https://github.com/pingcap/tidb/issues/44855) <span class="version-mark">从 v6.5.7 版本开始引入</span>
```
Comment (Oreoxmt) [inline optimizer-fix-controls.md:47]:
```suggestion
### [`44855`](https://github.com/pingcap/tidb/issues/44855) <span class="version-mark">从 v6.5.4 版本开始引入</span>
```

**Review by time-and-fate — APPROVED** [review]

File: system-variables.md [hunk 15]
```diff
@@ -2475,6 +2475,17 @@ mysql> desc select count(distinct a) from test.t;
 - 控制优化器是否会选择表的哈希连接。默认打开 (`ON`)。设置为 `OFF` 时，优化器在生成执行计划时会避免选择表的哈希连接，除非没有其他连接方式可用。
 - 如果同时使用了 `tidb_opt_enable_hash_join` 和 `HASH_JOIN` Hint，则 `HASH_JOIN` Hint 优先级更高。即使 `tidb_opt_enable_hash_join` 被设置为 `OFF`，如果在查询中指定了 `HASH_JOIN` Hint，TiDB 优化器仍然会强制执行哈希连接计划。
 
+### `tidb_opt_fix_control` <span class="version-mark">从 v6.5.3 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：是
```
Comment (time-and-fate) [inline system-variables.md:2482]:
```suggestion
```

**Review by ran-huang — APPROVED** [review]

**Pull request merged**